crates/ide-completion/src/completions/item_list/trait_impl.rs

```diff
@@ -265,10 +265,24 @@ fn add_type_alias_impl(
             };
 
             let start = transformed_ty.syntax().text_range().start();
-            let Some(end) = transformed_ty
-                .eq_token()
+
+            let end = if let Some(end) = transformed_ty
+                .syntax()
+                .children_with_tokens()
+                .find(|it| it.kind() == T![:])
                 .map(|tok| tok.text_range().start())
```

Comment:
```suggestion
            let end = if let Some(end) = transformed_ty.colon_token()
                .map(|tok| tok.text_range().start())
```

Reply:
This change needs a `HasTypeBounds` trait import

Reply:
thanks for your tips